lib/spack/spack/test/cmd/config.py

```diff
@@ -97,3 +114,67 @@ def test_config_list():
     output = config('list')
     assert 'compilers' in output
     assert 'packages' in output
+
+
+def test_config_update_packages(old_format_packages_yaml):
+    """Test Spack updating old packages.yaml format for externals
+    to new format. Ensure that data is preserved and converted
+    properly.
+    """
+    old_format_packages_yaml()
+    config('update', 'packages')
+
+    # Check the entries have been transformed
+    data = spack.config.get('packages')
+    assert 'externals' in data['cmake']
+    externals = data['cmake']['externals']
+    assert {'spec': 'cmake@3.14.0', 'prefix': '/usr'} in externals
+    assert 'externals' in data['gcc']
+    externals = data['gcc']['externals']
+    assert {'spec': 'gcc@8.3.0', 'module': 'gcc-8'} in externals
+
+
+def test_config_update_not_needed(mutable_config):
+    data_before = spack.config.get('repos')
+    config('update', 'repos')
+    data_after = spack.config.get('repos')
+    assert data_before == data_after
+
+
+def test_config_update_fail_if_bkp_is_there(old_format_packages_yaml):
+    # The first time it will update and create the backup file
+    old_format_packages_yaml()
+    config('update', 'packages')
+    # The second time it will stop because a backup file
+    # is already present
+    old_format_packages_yaml()
+    with pytest.raises(spack.main.SpackCommandError):
+        config('update', 'packages')
+
+
+def test_config_revert(old_format_packages_yaml):
+    cfg_file = old_format_packages_yaml()
+    bkp_file = cfg_file + '.bkp'
+
+    config('update', 'packages')
+
+    # Check that the backup file exists, compute its md5 sum
+    assert os.path.exists(bkp_file)
+    md5bkp = fs.md5sum(bkp_file)
+
+    config('revert', '--force', 'packages')
+
+    # Check that the backup file does not exist anymore and
+    # that the md5 sum of the configuration file is the same
+    # as that of the old backup file
+    assert not os.path.exists(bkp_file)
+    assert md5bkp == fs.md5sum(cfg_file)
```

Comment:
Python has a builtin module for checking file equality, no need to calculate checksums: https://docs.python.org/3/library/filecmp.html

Reply:
> Python has a builtin module for checking file equality, no need to calculate checksums:

1. Good to know as soon as we drop support for 2.6 and 2.7 :slightly_smiling_face: 
2. In this case the file is moved around, so I'm using the checksum to ensure that the content is what is expected

Reply:
1. This module is also in Python 2

Reply:
Completely correct, I misread the "new in Python 3.4" that refers only to the `clear_cache` function.